scripts/language_model/word_language_model.py

```diff
@@ -362,67 +373,105 @@ def train():
             target_list = gluon.utils.split_and_load(target, context, batch_axis=1, even_split=True)
             hiddens = detach(hiddens)
             Ls = []
-            L = 0
             with autograd.record():
                 for j, (X, y, h) in enumerate(zip(data_list, target_list, hiddens)):
-                    output, h, encoder_hs, dropped_encoder_hs = model.forward(X, h)
+                    output, h, encoder_hs, dropped_encoder_hs = model(X, h)
                     l = joint_loss(output, y, encoder_hs, dropped_encoder_hs)
-                    L = L + l.as_in_context(context[0]) / X.size
-                    Ls.append(l/X.size)
+                    Ls.append(l.as_in_context(context[0]) / (len(context) * X.size))
```

Comment:
`l` is already in the corresponding context, why do u copy it to the first gpu?

Reply:
The result should be the same, probably I can address later after KDD

Reply:
Let's make it right since this script is not used in KDD tutorials.

Reply:
The copy is not necessary in either single GPU or multi-GPU case.